test/functional/setup/ksp_app.c

```diff
@@ -0,0 +1,300 @@
+        SQLLEN cbdate;
+        rc = SQLBindParameter(stmt, 1, SQL_PARAM_INPUT, SQL_C_LONG, SQL_INTEGER, 0, 0, &c1, 0, 0);
+        checkRC(rc, "Binding parameters for insert", 9, stmt, SQL_HANDLE_STMT);
+        rc = SQLBindParameter(stmt, 2, SQL_PARAM_INPUT, SQL_C_CHAR, SQL_VARCHAR, 255, 0, c2, 255, 0);
+        checkRC(rc, "Binding parameters for insert", 9, stmt, SQL_HANDLE_STMT);
+        rc = SQLBindParameter(stmt, 3, SQL_PARAM_INPUT, SQL_C_CHAR, SQL_CHAR, 5, 0, c3, 5, 0);
+            checkRC(rc, "Binding parameters for insert", 9, stmt, SQL_HANDLE_STMT);
+        checkRC(rc, "Binding parameters for insert", 9, stmt, SQL_HANDLE_STMT);
+        cbdate = sizeof(SQL_DATE_STRUCT);
+        rc = SQLBindParameter(stmt, 4, SQL_PARAM_INPUT, SQL_C_TYPE_DATE, SQL_TYPE_DATE, 10, 0, &date, 0, &cbdate);   
+        checkRC(rc, "Binding parameters for insert", 9, stmt, SQL_HANDLE_STMT);
+        
+        date.year = 2017;
+        date.month = 8;
+        for (i = 0; i < 10; i++) {
+            date.day = i + 10;
+        
+            c1 = i * 10 + i + 1;
+            sprintf(c2, "Sample data %d for column 2", i);
+            for (j = 0; j < 3; j++) {
+                c3[j] = 'a' + i + j;
+            }
+            c3[3] = '\0';
+            rc = SQLExecDirect(stmt, "INSERT INTO CustomKSPTestTable (c1, c2, c3, c4) values (?, ?, ?, ?)", SQL_NTS);
+            checkRC(rc, "Inserting rows query", 10, stmt, SQL_HANDLE_STMT);
+        }
+        printf("(Encrypted) data has been inserted into CustomKSPTestTable. You may inspect the data now.\n");
+    }
+
+}
+
+int main(int argc, char **argv) {
+    char sqlbuf[1024];
+    SQLHENV env;
+    SQLHDBC dbc;
+    SQLHSTMT stmt;
+    SQLRETURN rc;
+    int i;    
+    char connStr[1024];
+    enum job task;
+    
+    if (argc < 6) {
+        fprintf(stderr, "usage: kspapp job server database uid pwd\n");
+        return 1;
+    }
+
+    task = atoi(argv[1]);
+    
+    sprintf(connStr, "DRIVER={ODBC Driver 13 for SQL Server};SERVER=%s;ColumnEncryption=Enabled;DATABASE=%s;UID=%s;PWD=%s", argv[2], argv[3], argv[4], argv[5]);
+   
+    /* Connect to Server */
+    rc = SQLAllocHandle(SQL_HANDLE_ENV, NULL, &env);
+    checkRC(rc, "allocating environment handle", 2, 0, 0);
+    rc = SQLSetEnvAttr(env, SQL_ATTR_ODBC_VERSION, (SQLPOINTER)SQL_OV_ODBC3, 0);
+    checkRC(rc, "setting ODBC version to 3.0", 3, env, SQL_HANDLE_ENV);
+    rc = SQLAllocHandle(SQL_HANDLE_DBC, env, &dbc);
+    checkRC(rc, "allocating connection handle", 4, env, SQL_HANDLE_ENV);
+    rc = SQLDriverConnect(dbc, 0, connStr, strlen(connStr), NULL, 0, NULL, SQL_DRIVER_NOPROMPT);
+    checkRC(rc, "connecting to data source", 5, dbc, SQL_HANDLE_DBC);
+    rc = SQLAllocHandle(SQL_HANDLE_STMT, dbc, &stmt);
+    checkRC(rc, "allocating statement handle", 6, dbc, SQL_HANDLE_DBC);
+
+    if (task == set_up) {
+        printf("Setting up KSP...\n");       
+        setKSPLibrary(stmt);
+        populateTestTable(dbc, stmt);
```

Comment:
Is there a reason why we're creating and populating a table here? wouldn't it be better to make and populate the tables on fly in the tests?

Reply:
I agree, but as of today I can only insert data into the test table with encryption using sqlsrv...
For now the KSP tests show that they can retrieve data from a pre-populated table with encryption

Reply:
hmmm this may change very soon though once the ODBC driver is ready. We shall see

Reply:
No we will keep using a test table because we want to show that we can retrieve data encrypted by a 3rd party (here in this case it's odbc )